providers/sugarcrm.go

```diff
@@ -0,0 +1,29 @@
+package providers
```

Comment:
can you put this file in a different PR? so have one PR for adding password grant and one PR for adding sugarCRM?

Reply:
Dedicated PR: https://github.com/amp-labs/connectors/pull/602